packages/http-client-csharp/emitter/src/type/input-type.ts

```diff
@@ -9,13 +9,33 @@ import {
   UsageFlags,
 } from "@azure-tools/typespec-client-generator-core";
 import { DateTimeKnownEncoding, DurationKnownEncoding } from "@typespec/compiler";
+import { InputOperation } from "./input-operation.js";
+import { InputParameter } from "./input-parameter.js";
 
-interface InputTypeBase {
+export interface InputClientType extends DecoratedType {
```

Comment:
Should this extend InputTypeBase?

Reply:
`InputTypeBase` was designed for model related stuff initially - let me see if it does not contain any extra stuffs on clients

Reply:
InputTypeBase has an extra field `deprecation` and client does not have that.

In typescript code, its interfaces are duck-typing - extending interfaces are only saving us time to write those common properties again which does not really represent anything.

We did not export `DecoratedType` nor `InputTypeBase` therefore I think it should be fine now.

Reply:
In that case, should we leave it named as `InputClient` to help differentiate? Also in C#, it is called `InputClient`. I think we should be consistent between the C# and TS type names unless there is good reason to diverge.

Reply:
sure, let me change it